pkg/apis/componentconfig/v1alpha1/defaults.go

```diff
@@ -98,7 +98,7 @@ func SetDefaults_KubeControllerManagerConfiguration(obj *KubeControllerManagerCo
 	if obj.HPAController.HorizontalPodAutoscalerCPUTaintPeriod == zero {
 		// Assuming CPU is collected every minute and initialization takes another minute HPA should
 		// disregard samples from first two minutes as contaminated by initialization.
-		obj.HPAController.HorizontalPodAutoscalerCPUTaintPeriod = metav1.Duration{Duration: 2 * time.Minute}
+		obj.HPAController.HorizontalPodAutoscalerCPUTaintPeriod = metav1.Duration{Duration: time.Minute}
```

Comment:
if this is based on HPA processing interval + window, and we are going to actually collect window from the API, why do we need this separate option?  Why not just use the HPA processing window, so we don't have this option at all?  (that's what I was trying to suggest before)

Reply:
We want to avoid using samples with window overlapping with pod initialization. If we use sample after {end of initialization time} + {sample window} we will achieve that.

{sample window} is hardcoded for now, I'll send PR for using actual value tomorrow.

We cannot know {end of initialization time} without storing per-pod information which is not feasible. So I'm going with a heuristic - initialization takes no more than a HorizontalPodAutoscalerCPUTaintPeriod (1 minute by default).